Foundation/NSJSONSerialization.swift

```diff
@@ -281,6 +291,21 @@ internal extension JSONSerialization {
     }
 }
 
+private protocol AnyOptional {
+    var value: Any? { get }
+}
+
+extension Optional : AnyOptional {
+    public var value: Any? {
+        switch self {
+        case .none:
+            return nil
+        case .some(let x):
+            return x
+        }
+    }
+}
+
```

Comment:
What's the benefit of putting this in a protocol here instead of just doing the switch at the couple of call sites where it's used? It feels one layer too abstracted to me.